### PR TITLE
Fix SDL omits URL during retry sequence

### PR DESCRIPTION
**Pull request by indlu**
Fix SDL omits one of the URLs while cyclically changing during retry sequence

Related defect APPLINK-31325

### CODE REVIEW

**Comment by indlu** [comment]
Please review : @dcherniev,  @LuxoftAKutsan, @dev-gh

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -1388,7 +1388,8 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
   }
 
   SendMessageToSDK(pt_string, urls.at(current_app).url.at(current_url));
-  current_url++;
+  if (!(urls.at(current_app).app_id == policy::kDefaultId))
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/policies/policy_handler.cc]: @indlu use breakets according gstyle

Comment (LuxoftAKutsan) [inline src/components/application_manager/src/policies/policy_handler.cc]: @indlu at can throw exeption, can't it?

Comment (LuxoftAKutsan) [inline src/components/application_manager/src/policies/policy_handler.cc]: @indlu 
```if (x != y)``` is better then ```if (!(x ==y))```

Comment (dev-gh) [inline src/components/application_manager/src/policies/policy_handler.cc]: @indlu 
What if 'default' will have more than one URL? I suppose only first of default will be used and other will be skipped.

**Comment by indlu** [comment]
Closed due to planned changes in functionality.

**Pull request closed without merging**